### PR TITLE
Fixes Cannot read property 'openChildren' of undefined

### LINKED ISSUE #1085: [ui] "Working..." splash stuck on Span view screen
I see this when running UI from the query service i.e. 1.37+

screen shot 2016-04-04 at 11 24 30 am

The console error trace looks like:

```
Uncaught TypeError: Cannot read property 'openChildren' of undefined
(anonymous function) @ trace.js:94
jQuery.extend.each @ jquery.js:365
(anonymous function) @ trace.js:92
jQuery.extend.each @ jquery.js:365
expandSpans @ trace.js:80
filterAdded @ trace.js:74
(anonymous function) @ trace.js:273
```

### PR DESCRIPTION
**Pull request by codefromthecrypt**
Before, the trace detail code would crash, trying to lookup an unknown
parent. This change tolerates missing parents and backfills tests around
the logic.

Fixes #1085

### CODE REVIEW

File: zipkin-ui/js/component_ui/trace.js [hunk 1]
```diff
@@ -3,6 +3,35 @@ import {component} from 'flightjs';
 import queryString from 'query-string';
 import $ from 'jquery';
 
+// extracted for testing. this code mutates spans and spansToShow
+export function showSpans(spans, parents, children, spansToShow) {
+  // mimic a Set
```
Comment (eirslett) [inline zipkin-ui/js/component_ui/trace.js]: We could load babel-polyfill and actually use a JavaScript Set?

File: zipkin-ui/test/component_ui/traceTestHelpers.js [hunk 2]
```diff
@@ -27,3 +27,23 @@ export function span(traceId,
     debug
   };
 }
+
+export function traceDetailSpan() {
+  return {openParents: 0, openChildren: 0, shown: false, show() {this.shown = true;}};
```
Comment (eirslett) [inline zipkin-ui/test/component_ui/traceTestHelpers.js]: Maybe use newlines here?

**Comment by eirslett** [comment]
Tests are great! :-D

File: zipkin-ui/test/component_ui/traceTestHelpers.js (hunk 2 above)
Comment (codefromthecrypt) [inline zipkin-ui/test/component_ui/traceTestHelpers.js]: done

File: zipkin-ui/js/component_ui/trace.js (hunk 1 above)
Comment (codefromthecrypt) [inline zipkin-ui/js/component_ui/trace.js]: done!

**Pull request merged**